File: packages/vm/wasmlib/go/wasmclient/clientview.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package wasmclient

type ClientView struct {
	svc *Service
	err error
	res ResMap
}

func (v *ClientView) Call(viewName string, args *Arguments) {
	if args == nil {
		args = &Arguments{}
	}
	v.res, v.err = v.svc.CallView(viewName, args.args)
}

func (v *ClientView) Error() error {
	return v.err
}

func (v *ClientView) Results() Results {
	return Results{res: v.res}
}

File: packages/vm/wasmlib/ts/wasmclient/clientview.ts
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import * as wasmclient from "./index"

export class ClientView {
    private svc: wasmclient.Service;

    constructor(svc: wasmclient.Service) {
        this.svc = svc;
    }

    protected async callView(viewName: string, args: wasmclient.Arguments | null, res: wasmclient.Results): Promise<void> {
        if (args == null) {
            args = new wasmclient.Arguments();
        }
        await this.svc.callView(viewName, args, res);
    }
}
